configuration/pih/liquibase/sql/fix_migrate_non_coded_mch_dxs.sql

```diff
@@ -0,0 +1,165 @@
+    ('BDCF non rassurant',concept_from_mapping('CIEL','168595')),
+    ('uc2 ancien',concept_from_mapping('CIEL','168594')),
+    ('anemie probable',concept_from_mapping('CIEL','121629')),
+    ('douleur epigastrique',concept_from_mapping('CIEL','141128')),
+    ('PES stab',concept_from_mapping('CIEL','168611'));
+
+-- sanity check, should be 0, do not continue if not
+select count(*) from dx_translations;
+
+-- get a count of the problematic obs, for reference
+select count(*) from obs coded where coded.concept_id=concept_from_mapping('PIH','DIAGNOSIS') and coded.value_coded is null and coded.voided=0;
+
+drop temporary table if exists coded_obs_to_recreate;
+create temporary table coded_obs_to_recreate
+(person_id                 int(11),
+ concept_id              int(11),
+ encounter_id              int(11),
+ order_id                    int(11),
+ obs_datetime              datetime,
+ location_id               int(11),
+ obs_group_id              int(11),
+ accession_number    varchar(255),
+ comments                    varchar(255),
+ previous_version          int(11),
+ non_coded_value   text,
+ value_coded             int(11)
+);
+
+insert into coded_obs_to_recreate (
+    person_id,
+    concept_id,
+    encounter_id,
+    order_id,
+    obs_datetime,
+    location_id,
+    obs_group_id,
+    accession_number,
+    comments,
+    previous_version,
+    non_coded_value)
+select
+    coded.person_id,
+    coded.concept_id,
+    coded.encounter_id,
+    coded.order_id,
+    coded.obs_datetime,
+    coded.location_id,
+    coded.obs_group_id,
+    coded.accession_number,
+    coded.comments,
+    coded.obs_id,
+    non_coded.value_text from obs coded, obs non_coded where
+        non_coded.concept_id=concept_from_mapping('PIH','Diagnosis or problem, non-coded') and non_coded.voided=1 and non_coded.void_reason='migrated to coded mch dx' and
+        coded.concept_id=concept_from_mapping('PIH','DIAGNOSIS') and coded.value_coded is null and coded.voided=0 and
+    non_coded.obs_group_id is not null and coded.obs_group_id is not null and non_coded.obs_group_id = coded.obs_group_id;
+
+-- figure out the coded obs to use for each
+update coded_obs_to_recreate u, dx_translations t set u.value_coded = t.code_dx where upper(trim(t.non_coded_dx)) = upper(trim(u.non_coded_value));
+
+-- record this number for sanity check later
+select count(*) from coded_obs_to_recreate;
+
+-- sanity check, should be 0, do not proceed if not
+select count(*) from coded_obs_to_recreate where value_coded is null;
+
+-- first void out the old obs that we are going recreate, to make sure if anything goes wrong we have the proper auditing infp
+update obs o, coded_obs_to_recreate coded set o.voided=1, o.void_reason='fixed migration to coded mch dx', o.voided_by =  (select user_id from users where username = 'admin'), o.date_voided=now()
+where coded.previous_version = o.obs_id;
+
+-- sanity check, should equal the count of rows in coded_obs_to_recreate
```

Comment:
Might want to run this script with autocommit = off and then explicitly commit at the end, that way these checks are actually meaningful after modifications have been made...